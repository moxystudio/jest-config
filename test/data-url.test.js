'use strict';

const fs = require('fs');
const path = require('path');
const tmp = require('tmp-sync');
const dataUrlTransformer = require('../addons/with-web/data-url');

const dir = tmp.in(__dirname);

afterAll(() => {
    tmp.clean();
});

it('should return string containing data-url', () => {
    const fileContent = 'this is a standard text file!';
    const filename = `${dir}/dataurl-test-file.txt`;

    // Can be verified at https://regex101.com/r/r8tE5u/3
    const matcher = /data:([\w/+]+);(charset=[\w-]+|base64),([a-zA-Z0-9+/]+={0,2})/;

    fs.writeFileSync(filename, fileContent);

    const output = dataUrlTransformer.process(null, filename);

    expect(output).toMatch(matcher);
    expect(output).toMatchSnapshot();
});

it('should throw when it cannot guess the MIME type of a file and show extension', () => {
    const fileContent = 'this is a test file with a strange extension!';
    const filename = `${dir}/dataurl-test-file.strange`;

    fs.writeFileSync(filename, fileContent);

    expect(() => dataUrlTransformer.process(null, filename)).toThrow(/Could not guess/);
    expect(() => dataUrlTransformer.process(null, filename)).toThrow(path.extname(filename));
});

it('should throw when it cannot guess the MIME type of a file and show whole filename when there is no extension', () => {
    const fileContent = 'this is a test file with a strange extension!';
    const filename = `${dir}/dataurl-test-file-strange`;

    fs.writeFileSync(filename, fileContent);

    expect(() => dataUrlTransformer.process(null, filename)).toThrow(/Could not guess/);
    expect(() => dataUrlTransformer.process(null, filename)).toThrow(path.basename(filename));
});
