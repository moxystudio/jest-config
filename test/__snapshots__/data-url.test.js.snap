// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return string containing data-url 1`] = `"module.exports = \\"data:text/plain;base64,dGhpcyBpcyBhIHN0YW5kYXJkIHRleHQgZmlsZSE=\\";"`;
